src/browser/modules/D3Visualization/lib/visualization/renders/menu.ts

```diff
@@ -102,42 +100,51 @@ const createMenuItem = function(
     .classed(className, true)
     .classed('context-menu-item', true)
     .attr({
-      d(node: any) {
+      d(node: NodeVisualisationModel) {
         // @ts-expect-error Expected 1-2 arguments, but got 0.ts(2554)
-        return arc(node.radius, itemNumber, 1)()
+        return drawArc(node.radius, itemIndex, 1)()
       }
     })
 
-  const rawSvgIcon = icons[textValue]
+  const rawSvgIcon = icons[svgIconKey]
+  const svgIcon = document.importNode(
+    new DOMParser().parseFromString(rawSvgIcon, 'application/xml')
+      .documentElement.firstChild as HTMLElement,
+    true
+  )
   const icon = iconPath
     .enter()
-    .appendSVG(rawSvgIcon)
+    .append('g')
+    .html(svgIcon.innerHTML)
+    .classed('icon', true)
```

Comment:
This is the core change for implementing `appendSVG` method that was removed in this PR.
The rest of the diff is code cleanups and refactors.